chain/jsonrpc/res/epoch_info.html

```diff
@@ -71,6 +71,48 @@
             return bytes.toFixed(dp) + ' ' + units[u];
         }
 
+        function process_validator_info(validator_info) {
+            let total_stake = 0;
+            validator_info.current_validators.forEach(validator => {
+                total_stake += parseInt(validator.stake);
+            })
+            $('.js-tbody-curepoch').append("Total stake " + total_stake)
+            validator_info.current_validators.forEach(validator => {
+                let row = $('<tr>');
+                console.log(validator.stake + "validator stake\n");
+                row.append($('<td>').append(validator.account_id));
+                row.append($('<td>').append((100 * validator.stake / total_stake).toFixed(2) + "%"));
+                row.append($('<td>').append(validator.shards.join(",")));
+                row.append($('<td>').append(validator.num_expected_blocks - validator.num_produced_blocks));
+                row.append($('<td>').append(validator.num_expected_blocks));
+                row.append($('<td>').append(validator.num_expected_chunks - validator.num_produced_chunks));
+                row.append($('<td>').append(validator.num_expected_chunks));
+                $('.js-tbody-curepoch-curvalidators').append(row)
+            })
+
+            let row1 = $('<tr>');
+            let row2 = $('<tr>');
+            row1.append("<th> Account Id </th>");
+            row2.append("<th> Stake / Stake Percentage </th>");
+            validator_info.current_proposals.forEach(proposal=> {
+                row1.append($('<td>').append(proposal.account_id));
+                row2.append($('<td>').append(proposal.stake + "/" + proposal.stake/total_stake.toFixed(2) + "%"));
+            })
+            $('.js-tbody-curepoch-proposals').append(row1).append(row2);
+
+            row1 = $('<tr>');
+            row2 = $('<tr>');
+            row1.append("<th> Account Id </th>");
+            row2.append("<th> Kickout Reason </th>");
+            validator_info.prev_epoch_kickout.forEach(kickout=> {
```

Comment:
this can get very wide.. (as you're adding a column per each validator that got kicked out, right?)

Reply:
maybe let's make it still a 'one validator per row' -- but  in such a way that we can 'fold' this whole section.